### PR TITLE
More fuzzy sub-auto

### PR DESCRIPTION
**Pull request by kwkam**
Split the name of both files, by some common delimiters, into parts and compare them
eg. `"xx - yyyy"`, `"(xx) yyyy"`, `"[xx][yyyy]"` will be broken into `{ "xx", "yyyy" }`

### CODE REVIEW

**Comment by wm4** [comment]
That's really complex and I'm not particularly convinced that it helps with anything.

File: misc/bstr.c [hunk 1]
```diff
@@ -100,7 +100,8 @@ int bstrspn(struct bstr str, const char *accept)
 
 int bstr_find(struct bstr haystack, struct bstr needle)
 {
-    for (int i = 0; i < haystack.len; i++)
```
Comment (kwkam) [inline misc/bstr.c]: `bstr_startswith(struct bstr str, struct bstr prefix)` will always return `false` if `str.len < prefix.len`
When length of `bstr_splice(haystack, i, haystack.len)` is less than `needle`, this function should return already

File: misc/bstr.c [hunk 2]
```diff
@@ -100,7 +100,8 @@ int bstrspn(struct bstr str, const char *accept)
...
+    size_t n = haystack.len > needle.len ? haystack.len - needle.len + 1 : 0;
+    for (int i = 0; i < n; i++)
         if (bstr_startswith(bstr_splice(haystack, i, haystack.len), needle))
```
Comment (kwkam) [inline misc/bstr.c]: `bstr_startswith(struct bstr str, struct bstr prefix)` will always return `false` if `str.len < prefix.len`
When length of `bstr_splice(haystack, i, haystack.len)` is less than `needle`, this function should return already

**Comment by wm4** [comment]
To get this accepted, you should either give a convincing argument why this would be needed, and/or simplify the code as far as possible. Otherwise I'll reject it.

**Comment by otommod** [comment]
I too agree that this has no place in mpv core.  Can't you do this via lua perhaps?

**Comment by kwkam** [comment]
Sorry for the late reply.

to wm4:
I don't really know whether this is necessary, but it solves my problem. I suggest it here because I think it would not be a bad feature to have. Maybe create a `veryfuzzy` option for it so the current behaviour will not be changed.
Regarding the codes, I have tried to make it better. Since I'm not a coder, I don't know things beyond it works or not. If the code is still too bad, or this feature is not needed, you may close this request.

to otommod:
I'm not sure because I don't know how mpv works with lua. If the maintainer suggests so, I might dig into the code to see whether I could do it.

Thank you for your attention.

**Comment by enoposix** [comment]
Can this be closed?

**Pull request closed without merging**